runtime/src/stakes.rs

```diff
@@ -489,6 +503,47 @@ pub(crate) mod serde_stakes_enum_compat {
     }
 }
 
+fn refresh_vote_accounts(
+    thread_pool: &ThreadPool,
+    epoch: Epoch,
+    vote_accounts: &VoteAccounts,
+    stake_delegations: &[&StakeAccount],
+    stake_history: &StakeHistory,
+) -> VoteAccounts {
+    type StakesHashMap = HashMap</*voter:*/ Pubkey, /*stake:*/ u64>;
+    fn merge(mut stakes: StakesHashMap, other: StakesHashMap) -> StakesHashMap {
+        if stakes.len() < other.len() {
+            return merge(other, stakes);
+        }
+        for (pubkey, stake) in other {
+            *stakes.entry(pubkey).or_default() += stake;
+        }
+        stakes
+    }
+    let stake_history = Some(stake_history.deref());
+    let delegated_stakes = thread_pool.install(|| {
+        stake_delegations
+            .par_iter()
+            .fold(HashMap::default, |mut delegated_stakes, stake_account| {
+                let delegation = stake_account.delegation();
+                let entry = delegated_stakes.entry(delegation.voter_pubkey).or_default();
+                *entry += delegation.stake(epoch, stake_history);
+                delegated_stakes
+            })
+            .reduce(HashMap::default, merge)
+    });
+    vote_accounts
+        .iter()
+        .map(|(&vote_pubkey, vote_account)| {
+            let delegated_stake = delegated_stakes
+                .get(&vote_pubkey)
+                .copied()
+                .unwrap_or_default();
+            (vote_pubkey, (delegated_stake, vote_account.clone()))
+        })
+        .collect()
+}
```

Comment:
I can't see any differences, but to double-check, this function is just refactoring, correct?

Reply:
Yes, this is the common code between `activate_epoch` and the newly added `update_stake_accounts` factored out here.